Bottom: 7918641981ba21627467828a2bb3942aa2ec2f70
Top:    cecefe95177c7cf3ddad81c213249aec93b601bd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-30 15:01:13 +0100

djopt.c: Tidy up ActionDJopt() using LOOP macros


---

diff --git a/src/djopt.c b/src/djopt.c
index 842ed47..8e94d7e 100644
--- a/src/djopt.c
+++ b/src/djopt.c
@@ -3043,20 +3043,18 @@ ActionDJopt (int argc, char **argv, int x, int y)
   for (layn = 0; layn < max_copper_layer; layn++)
     {
       LayerType *layer = LAYER_PTR (layn);
-      int ln;
-      for (ln = 0; ln < layer->LineN; ln++)
+
+      LINE_LOOP (layer);
 	{
-	  LineType *l = &(layer->Line[ln]);
 	  line_s *ls;
 
 	  /* don't mess with thermals */
-	  if (TEST_FLAG (USETHERMALFLAG, l))
+	  if (TEST_FLAG (USETHERMALFLAG, line))
 	    continue;
 
-	  if (l->Point1.X == l->Point2.X && l->Point1.Y == l->Point2.Y)
+	  if (line->Point1.X == line->Point2.X && line->Point1.Y == line->Point2.Y)
 	    {
-	      RemoveLine (layer, l);
-	      ln--;
+	      RemoveLine (layer, line);
 	      continue;
 	    }
 
@@ -3064,13 +3062,14 @@ ActionDJopt (int argc, char **argv, int x, int y)
 	  ls->next = lines;
 	  lines = ls;
 	  ls->is_pad = 0;
-	  ls->s = find_corner (l->Point1.X, l->Point1.Y, layn);
-	  ls->e = find_corner (l->Point2.X, l->Point2.Y, layn);
-	  ls->line = l;
+	  ls->s = find_corner (line->Point1.X, line->Point1.Y, layn);
+	  ls->e = find_corner (line->Point2.X, line->Point2.Y, layn);
+	  ls->line = line;
 	  add_line_to_corner (ls, ls->s);
 	  add_line_to_corner (ls, ls->e);
 	  ls->layer = layn;
 	}
+      END_LOOP;
     }
